### ISSUE TITLE
Enable NiftiLabelsMasker to provide un-aggregated values Or NiftiMasker to provide resample target

### ISSUE THREAD

**Issue by DasDominus** [nilearn/nilearn#2686]
currently nilearn.input_data.NiftiLabelsMasker provides aggregation strategies.
However, it does not have the option to simply get the raw signal without aggregation.

### Benefits to the change
It would enable more flexibility when performing analysis

### Pseudocode for the new behavior, if applicable

in https://github.com/nilearn/nilearn/blob/e372946bfc91419bc1dbbfadebfed6e6a402956d/nilearn/input_data/nifti_labels_masker.py#L32

instead of using only:

-          signal_extraction.img_to_signals_labels 

have the option of 

- using img_to_signals_maps

**Comment by bthirion** [comment]
But this means that it would exactly have the behavior of NiftiMasker ? Why not simply use NiftiMasker ?

**Comment by DasDominus** [comment]
NiftiLabelsMasker for example provides resample function based on data.
NiftiMasker however, resamples to the mask first unless target_shape & target_affine from data are provided.

I was mentioning to extend NiftiLabelsMasker because this seems simpler. But for the sake of what I'm trying to do, providing resampling_target to NiftiMasker would do the trick as well.

**Comment by tsalo** [comment]
I could be misunderstanding how `NiftiMasker` could be used to solve the original problem, but I think having the ability to return unaggregated values, split by label, in `NiftiLabelsMasker` would at least be useful for carpet plots.

**Comment by bthirion** [comment]
My point was that `NiftiMasker` indeed returns unaggregated values for the provided regions, so you cna compute any thing you want on these values afterwards.
I am reluctant to changing the behavior of `NiftiLabelsMasker` by creating the option to return unaggregated values. this makes theses objects even harder to understand.

**Comment by DasDominus** [comment]
Thanks @bthirion ! Yeah that do make things a bit harder to understand.
I eventually just used NiftiMasker and manually did aggregations

That being said, NiftiLabelsMasker do provide an option strategy which supports:

Must be one of: sum, mean, median, mininum, maximum, variance,
        standard_deviation. Default='mean'

which eventually resolves to getting functions from the scipy.ndimage package

Enabling customization of this option seems to be too complex for most users. But for the sake of completeness,

for those wondering how to pass in custom reduction function, one way is to modify package locally and replace strategy with a lambda reduction function.

**Comment by fraimondo** [comment]
> But this means that it would exactly have the behavior of NiftiMasker ? Why not simply use NiftiMasker ?

I'm trying to achieve a similar behaviour here. The problem is that I have an atlas (e.g. Schaefer 100 rois) and I want to have the unaggregated values for each ROI) to later on apply a user-defined aggregation.

I've been trying to understand how is this achievable using NiftiMasker as mentioned above.

This example gives me the mean GM density per ROI:

```Python
from nilearn import datasets
from nilearn.maskers import NiftiMasker, NiftiLabelsMasker
oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=1)

vbm = oasis_dataset.gray_matter_maps[0]
atlas = datasets.fetch_atlas_schaefer_2018(n_rois=100)

nifti_masker = NiftiLabelsMasker(labels=atlas.maps)
mean_gm = nifti_masker.fit_transform(vbm)
```

Now I would like to apply a custom `strategy` (e.g. trimmed mean, winsorized mean, etc). How do you think of achieving this with the NiftiMasker? You mean applying the NiftiMasker for each label? (I believe this is computationally very heavy due to checking all the image/mask attributes every time)

**Comment by fraimondo** [comment]
So I followed this approach and did a couple of tests.

This is using the NiftiMasker to compute the mean:

```Python
    # Get the oasis VBM data:
    oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=1)
    vbm = oasis_dataset.gray_matter_maps[0]
    img = nib.load(vbm)

    # Mask atlas manually
    atlas_img_res = resample_to_img(
        atlas.maps,
        img,
        interpolation='nearest',
    )
    atlas_bin = math_img(
        'img != 0',
        img=atlas_img_res,
    )

    masker = NiftiMasker(atlas_bin, target_affine=img.affine)

    data = masker.fit_transform(img)
    atlas_values = masker.transform(atlas_img_res)
    atlas_values = np.squeeze(atlas_values).astype(int)

    # Compute the mean manually
    manual = []
    for t_v in sorted(np.unique(atlas_values)):
        t_values = np.mean(data[:, atlas_values == t_v])
        manual.append(t_values)
    manual = np.array(manual)[np.newaxis, :]
```

And this is using the NiftiLabelsMasker to compute the mean.

```Python
nifti_masker = NiftiLabelsMasker(labels_img=atlas.maps)
auto = nifti_masker.fit_transform(vbm)
```

My problem now is that these results are not equal. I'm trying to understand what might be the difference between using the NiftiLabelsMasker and the NiftiMasker, as conceptually they should to the same. Any thoughts?

**Comment by bthirion** [comment]
When I try to replicate your example, I get the same results. Which atlas are you using ?

**Comment by fraimondo** [comment]
Never mind, I had a bug in my code and in the example I used `assert_array_equal` instead of `assert_array_almost_equal`.
